### ISSUE TITLE
sorting friends

### ISSUE THREAD

**Issue by jancborchardt** [agical/friendsunhosted#6]
The friends list should sort the friends automatically by some sort of activity algorithm which combines:
number-of-posts + most-recent-post + friends-in-common

**Comment by brolund** [comment]
Nice idea. For the friends of friends that could mean fetching a lot of data from the servers, unless we aggregate the information automatically in the public profile, which gives us the cache invalidation problem... ;-)

**Comment by johlrogge** [comment]
Perhaps we can update some data when posting such as "last post", "post frequency" etc. Then we just have to calculate frends in common for which we need to fetch data anyway. Basically: calculate once, read many

**Comment by jancborchardt** [comment]
Yeah, we could first start doing it for friends only. Then for friends of friends later, because for them the data is mostly important to recommend you new people to friend (which you might have chosen not to do).

**Comment by johlrogge** [comment]
I would actually prefer to have my friends sorted alphabetically. Though suggestions for people I "might know" would be nice to have sorted in order of probability.

It would also be nice to have an option to see who is more active etc of my friends.

**Comment by jancborchardt** [comment]
Friends you don’t interact with often (below a specific threshold) should be sorted alphabetically, yes. But active friends and people you interact with often (add »comments on each others posts« to the algorithm) are probably just 5 or 10, and they should automatically be sorted up top.

Initially we should do with as little options or different views as possible and instead look for a good single way to design it.

**Comment by johlrogge** [comment]
Sounds good
